rest/attachment_migration_test.go

```diff
@@ -0,0 +1,455 @@
+	base.SetUpTestLogging(t, base.LevelInfo, base.KeyAll)
+	tb := base.GetTestBucket(t)
+	rtConfig := &RestTesterConfig{
+		CustomTestBucket: tb,
+		PersistentConfig: true,
+	}
+
+	rt := NewRestTesterMultipleCollections(t, rtConfig, 2)
+	defer rt.Close()
+	ctx := base.TestCtx(t)
+	_ = rt.Bucket()
+
+	const (
+		dbName         = "db1"
+		totalDocsAdded = 8000
+	)
+
+	ds0, err := tb.GetNamedDataStore(0)
+	require.NoError(t, err)
+	ds1, err := tb.GetNamedDataStore(1)
+	require.NoError(t, err)
+	opts := &sgbucket.MutateInOptions{}
+
+	// add some docs (with xattr so they won't be ignored in the background job) to both collections
+	bodyBytes := []byte(`{"some": "body"}`)
+	for i := 0; i < 4000; i++ {
+		key := fmt.Sprintf("%s_%d", t.Name(), i)
+		xattrsInput := map[string][]byte{
+			"_xattr": []byte(`{"some":"xattr"}`),
+		}
+		_, writeErr := ds0.WriteWithXattrs(ctx, key, 0, 0, bodyBytes, xattrsInput, nil, opts)
+		require.NoError(t, writeErr)
+	}
+	for i := 0; i < 4000; i++ {
+		key := fmt.Sprintf("%s_%d", t.Name()+"greg", i)
+		xattrsInput := map[string][]byte{
+			"_xattr": []byte(`{"some":"xattr"}`),
+		}
+		_, writeErr := ds1.WriteWithXattrs(ctx, key, 0, 0, bodyBytes, xattrsInput, nil, opts)
+		require.NoError(t, writeErr)
+	}
+
+	scopesConfigC1Only := GetCollectionsConfig(t, tb, 2)
+	dataStoreNames := GetDataStoreNamesFromScopesConfig(scopesConfigC1Only)
+	scope := dataStoreNames[0].ScopeName()
+	collection1 := dataStoreNames[0].CollectionName()
+	collection2 := dataStoreNames[1].CollectionName()
+	delete(scopesConfigC1Only[scope].Collections, collection2)
+
+	scopesConfigBothCollection := GetCollectionsConfig(t, tb, 2)
+
+	// Create a db1 with one collection initially
+	dbConfig := rt.NewDbConfig()
+	dbConfig.AutoImport = false
+	dbConfig.Scopes = scopesConfigC1Only
+
+	resp := rt.CreateDatabase(dbName, dbConfig)
+	RequireStatus(t, resp, http.StatusCreated)
+
+	dbCtx := rt.GetDatabase()
+	mgr := dbCtx.AttachmentMigrationManager
+	assert.Len(t, dbCtx.RequireAttachmentMigration, 1)
+	scNames := base.ScopeAndCollectionNames{}
+	scName := base.ScopeAndCollectionName{Scope: scope, Collection: collection1}
+	scNames = append(scNames, scName)
+	assert.ElementsMatch(t, scNames, dbCtx.RequireAttachmentMigration)
+	// wait for migration job to start
+	db.RequireBackgroundManagerState(t, ctx, mgr, db.BackgroundProcessStateRunning)
+
+	// update db config to include second collection
+	dbConfig = rt.NewDbConfig()
+	dbConfig.AutoImport = false
+	dbConfig.Scopes = scopesConfigBothCollection
+	resp = rt.UpsertDbConfig(dbName, dbConfig)
+	RequireStatus(t, resp, http.StatusCreated)
+
+	// wait for attachment migration job to start and finish
+	dbCtx = rt.GetDatabase()
+	mgr = dbCtx.AttachmentMigrationManager
+	assert.Len(t, dbCtx.RequireAttachmentMigration, 2)
+	scName = base.ScopeAndCollectionName{Scope: scope, Collection: collection2}
+	scNames = append(scNames, scName)
+	assert.ElementsMatch(t, scNames, dbCtx.RequireAttachmentMigration)
```

Comment:
```suggestion
	scNames = append(scNames, base.ScopeAndCollectionName{Scope: scope, Collection: collection2})
	assert.ElementsMatch(t, scNames, dbCtx.RequireAttachmentMigration)
```

Then we will see the expected and difference right away.